### ISSUE TITLE
HLS Live Streaming on IOS

### ISSUE THREAD

**Issue by ericlifs** [TheWidlarzGroup/react-native-video#2337]
# Bug
I want to configure the buffer for live streaming videos on iOS but the readme says that the bufferConfig prop works only for android. Also, the already existing ways of getting the available bandwidth updates are not working on ios.

Not sure if there is a workaround for setting/getting those values.

## Platform
Which player are you experiencing the problem on:
* iOS

## Environment info
System:
    OS: macOS 10.15.7
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 25.18 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.16.3 - ~/.nvm/versions/node/v12.16.3/bin/node
    Yarn: 1.22.10 - ~/.nvm/versions/node/v12.16.3/bin/yarn
    npm: 6.14.4 - ~/.nvm/versions/node/v12.16.3/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.1 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.2, DriverKit 20.0, macOS 11.0, tvOS 14.2, watchOS 7.1
    Android SDK: Not Found
  IDEs:
    Android Studio: Not Found
    Xcode: 12.2/12B45b - /usr/bin/xcodebuild
  Languages:
    Java: Not Found
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: Not Found
    react-native: Not Found
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

Library version: 5.0.4

## Video sample
https://test-streams.mux.dev/x36xhzz/x36xhzz.m3u8

**Comment by ApWin** [comment]
Same issue (

**Comment by ApWin** [comment]
@ericlifs do you have any solution for this ?

**Comment by ericlifs** [comment]
@ApWin no, not at all. I'm hoping someone from react-native-video can provide more info on this.

**Comment by scottschindler** [comment]
Same issue. Any luck @ericlifs or @ApWin ?

**Comment by TuannTran04** [comment]
Have you done it yet? Please show me how to do it

**Comment by Michota** [comment]
We need it so badly!

**Comment by freeboub** [comment]
Here is a start of implementation I reopen yesterday:  https://github.com/react-native-video/react-native-video/pull/1353 not sure how it goes ...

**Comment by DLowHP** [comment]
Does anybody here know if it is possible to set a single native player property while keeping the react-native-player intact? Maybe it's possible to patch it yourself without changing the library?

**Comment by freeboub** [comment]
> Does anybody here know if it is possible to set a single native player property while keeping the react-native-player intact? Maybe it's possible to patch it yourself without changing the library? 

Maybe you can open a discussion, but the answer will be that both exoplayer shall have the same version...

**Comment by Michota** [comment]
Bump. I need this so badly. I would even move from Expo-AV and donate some cash if iOS buffer config will be working as intended.

**Comment by KrzysztofMoch** [comment]
@Michota it seams that AVPlayer (iOS) is very limited in this area compared to android - do you have any specific parameter that you would have ?
